session/payment/session_payments.go

```diff
@@ -124,6 +136,26 @@ func (cpo *SessionPayments) validateBalanceDifference(balance uint64) error {
 	return nil
 }
 
+func (cpo *SessionPayments) sendWithRetry(send func() error, attempts int, delay time.Duration) error {
```

Comment:
I dont see this retry as necessary because promised are already repeated each Xs
I see this retry as retry of retries :)